--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -27,6 +_,11 @@
         if (System.getProperty("joml.nounsafe") == null) {
             System.setProperty("joml.nounsafe", "true");
         }
+        // Velvety start - Velvety config
+        String sparkExtraConfigs = System.getProperty("spark.serverconfigs.extra");
+        String velvetyConfigs = com.kacimiamine.velvety.config.VelvetyConfig.getSparkVelvetyConfigs();
+        System.setProperty("spark.serverconfigs.extra", sparkExtraConfigs == null ? velvetyConfigs : sparkExtraConfigs + "," + velvetyConfigs);
+        // Velvety end - Velvety config
 
         OptionParser parser = new OptionParser() {
             {
@@ -143,6 +_,14 @@
                         .ofType(File.class)
                         .defaultsTo(new File[] {})
                         .describedAs("Plugin directory");
+
+                // Velvety start - Velvety config
+                this.acceptsAll(asList("velvety-dir", "velvety-settings-directory"), "Directory for Velvety settings")
+                    .withRequiredArg()
+                    .ofType(File.class)
+                    .defaultsTo(new File(com.kacimiamine.velvety.config.VelvetyConfig.CONFIG_DIR_NAME))
+                    .describedAs("Config directory");
+                // Velvety end - Velvety config
 
                 this.accepts("server-name", "Name of the server")
                         .withRequiredArg()
