--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1280,7 +_,7 @@
             // Paper end - Add onboarding message for initial server start
             // Paper start - Improve outdated version checking
             if (System.getProperty("paper.disableStartupVersionCheck") == null && io.papermc.paper.configuration.GlobalConfiguration.get().updateChecker.enabled) {
-                CompletableFuture.runAsync(com.destroystokyo.paper.PaperVersionFetcher::getUpdateStatusStartupMessage, io.papermc.paper.util.MCUtil.ASYNC_EXECUTOR);
+                CompletableFuture.runAsync(com.kacimiamine.velvety.util.VelvetyVersionFetcher::getUpdateStatusStartupMessage, io.papermc.paper.util.MCUtil.ASYNC_EXECUTOR); // Velvety - Rebrand
             }
             // Paper end - Improve outdated version checking
 
